### PR TITLE
Emergency shuttle panel timer

### LINKED ISSUE #24408: Changing the timer via the "emergency shuttle panel" sends the announcement again
#### Steps to reproduce
- Call the shuttle
- Change the ETA
- Get a second big PRIORITY ANNOUNCEMENT "The emergency shuttle has been called."
- now everybody knows you're a shitmin

### PR DESCRIPTION
**Pull request by Zth--**
Closes #24408

The code speaks for itself. Adds more behaviour to editing the timer on the _emergency shuttle panel_

### CODE REVIEW

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -356,9 +356,18 @@
...
 			return
 
-		emergency_shuttle.settimeleft( input("Enter new shuttle duration (seconds):","Edit Shuttle Timeleft", emergency_shuttle.timeleft() ) as num )
+		var/new_timeleft = input("Enter new shuttle duration (seconds):","Edit Shuttle Timeleft", emergency_shuttle.timeleft() ) as num | null
+		if(!new_timeleft)
+			return
+
+		if( alert("Do you want this to be announced?",,"Yes","No" ) == "Yes" )
+			if(new_timeleft < emergency_shuttle.timeleft())
+				captain_announce("The emergency shuttle found a shortcut. It will arrive in [round(new_timeleft/60)] minutes.")
```
Comment (Kurfursten) [inline code/modules/admin/topic.dm]: Could you come up with something a little more logical for a spacefaring vessel?

Even as simple as
"is arriving ahead of schedule"
"hit the turbo"
"has been scrambled for faster deployment"

Comment (utilitas-) [reply code/modules/admin/topic.dm]: "has engaged nitro afterburners" is clearly the best option here.

Comment (Zth--) [reply code/modules/admin/topic.dm]: keep them coming, I'm adding all of them

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -356,9 +356,24 @@
 		if(!check_rights(R_SERVER))
 			return
 
-		emergency_shuttle.settimeleft( input("Enter new shuttle duration (seconds):","Edit Shuttle Timeleft", emergency_shuttle.timeleft() ) as num )
+		var/new_timeleft = input("Enter new shuttle duration (seconds):","Edit Shuttle Timeleft", emergency_shuttle.timeleft() ) as num | null
+		if(!new_timeleft)
+			return
+
+		var/reason
+		if( alert("Do you want this to be announced?",,"Yes","No" ) == "Yes" )
```
Comment (DamianX) [inline code/modules/admin/topic.dm]: Another chance to cancel here would be nice

**Comment by Kurfursten** [comment]
Give the man his cancel and we'll get this baby merged

**Comment by Zth--** [comment]
Will do tonight

**Comment by Zth--** [comment]
Done

**Pull request merged**